### ISSUE TITLE
No updates until Jan 2025

### ISSUE THREAD

**Issue by satanTime** [help-me-mom/ng-mocks#10487]
I guess you all noticed no active development of the library for several years. 

It's going to change from January 2025. 

Thanks to all who was patient and hasn't lost trust in ng-mocks.

**Comment by JPtenBerge** [comment]
Not to add to any stress, but will Angular 19 get a relatively quick version support release before that time by any chance?

**Comment by evtk** [comment]
Thanks for the update @satanTime. I still use ng-mocks on daily basis as it abstracts so much of the unit test fuzz. 

I came here after watching the updates for NG19 and noticing how there is getting more and more attention to implementing signals. The simple fact is that ng-mocks currently prevents me from starting to use the signal input api for instance instead of the regular Input decorator. 

So really looking forward to updates on that part!

**Comment by JPtenBerge** [comment]
@evtk ng-mocks hasn't prevented me from using signal inputs.

```ts
export class InterviewStepsComponent {
    step = input.required<Step>();
```

```ts
fixture = TestBed.createComponent(InterviewStepsComponent);
sut = fixture.componentInstance;
fixture.componentRef.setInput('step', currentStep);
```

I'm not using `MockRender()` though, is that what you're using?

**Comment by GipHub123** [comment]
@evtk 
I can confirm that input signals work with the current version of ngMocks. However, as far as I know, certain features, such as `ContentChildren `and `ViewChild`, are not yet supported.
 https://github.com/help-me-mom/ng-mocks/issues/8895

**Comment by evtk** [comment]
> @evtk ng-mocks hasn't prevented me from using signal inputs. ng-mocks simply doesn't provide any help in that area.
> 
> ```ts
> export class MyStepComponent {
>     step = input.required<Step>();
> ```
> 
> ```ts
> beforeEach(() => {
>     TestBed.configureTestingModule({
>         providers: [MockProvider(...), MockProvider(...)],
>         imports: [MyStepComponent],
>     });
> 
>     fixture = TestBed.createComponent(MyStepComponent);
>     sut = fixture.componentInstance;
>     fixture.componentRef.setInput('step', currentStep);
> });
> ```
> 
> I'm not using `MockRender()` though, is that what you're using?

Yeah definitely went for being `Pro` and using `MockRender` everywhere in my tests :|

**Comment by JPtenBerge** [comment]
@satanTime Understood, thanks for the response.

**Comment by dmitry-stepanenko** [comment]
@jahusa02 if you will work on v19 support, you might be interested to look into my fork https://github.com/dmitry-stepanenko/ng-mocks/tree/ds/v19 . I managed to get standalone components to work well in v19 apps (it's a breaking change though). Package version with my changes can be installed directly from https://github.com/dmitry-stepanenko/ng-mocks/tree/v19-built

The only thing that I did not figure out is how to differentiate between v19 modules and old pre-compiled ones that come from 3rd party packages, because old ones should still keep "standalone: false" by default.

**Comment by GipHub123** [comment]
Hmmmh. If I'm not mistaken, 19 version fix seemed to be quite trivial 🤔 


**func.is-standalone.ts**

`return collectDeclarations(declaration)[type].standalone === true;`
-->
`return collectDeclarations(declaration)[type].standalone !== false;`

@dmitry-stepanenko I didn’t find any version-related code either, so I assume that metadata is being used instead

In another words, change is not backward compatible.

**Comment by dmitry-stepanenko** [comment]
@GipHub123 if the mocked component imports an npm package that don't use standalone components inside, test will fail

**Comment by re-alam** [comment]
@GipHub123 @dmitry-stepanenko 
I've found a simpler fix to make `MockRender` compatible with Angular 19. By setting the standalone property to false when generating wrapper component fixes the issue. I've also updated the isStandalone method to ensure backward compatibility. Please let me know if it works for you.

**Comment by GipHub123** [comment]
@re-alam I reviewed the code and it's seemed to be solid / it's backward compatible (I didn't test it) 👍 Version check seemed to be quite trivial / elegant 👍

**Comment by thiboot** [comment]
Good news for january !

**Comment by florestankorp** [comment]
> @GipHub123 @dmitry-stepanenko I've found a simpler fix #10583 to make `MockRender` compatible with Angular 19. By 😎setting the standalone property to false when generating wrapper component fixes the issue. I've also updated the isStandalone method to ensure backward compatibility. Please let me know if it works for you.

Bro joined GitHub 2 weeks ago, forks 'ng-mocks', opens a PR with a fix for Angular v19. Absolute legend 😎

**Comment by Tallyb** [comment]
vitest support would be great. Happy new year!

**Comment by Martinspire** [comment]
> Hmmmh. If I'm not mistaken, 19 version fix seemed to be quite trivial 🤔
> 
> **func.is-standalone.ts**
> 
> `return collectDeclarations(declaration)[type].standalone === true;` --> `return collectDeclarations(declaration)[type].standalone !== false;`
> 
> @dmitry-stepanenko I didn’t find any version-related code either, so I assume that metadata is being used instead
> 
> In another words, change is not backward compatible.

Couldn't there just be an iffy returning depending on what angular version is being used?

**Comment by GipHub123** [comment]
@Martinspire 

`Couldn't there just be an iffy returning depending on what angular version is being used?`

If you review @re-alam 's  code, you'll find the version check you were looking for.

**Comment by Martinspire** [comment]
So can we merge that? Its January after all 😄

**Comment by endlacer** [comment]
I think we have to rethink my approach to using libs from now on. Several of our large projects are blocked from being upgraded bc of this simple PR simply wont get merged :(

**Comment by Tallyb** [comment]
@endlacer patch-package is your friend.

**Comment by GipHub123** [comment]
@endlacer 

`I think we have to rethink my approach to using libs from now on. Several of our large projects are blocked from being upgraded bc of this simple PR simply wont get merged`

Would it make more sense to reconsider your update policy? 😐
Why rush to update a production application immediately after a new Angular major version is released, potentially putting yourself in a difficult position?
In my opinion, it’s wiser to wait until at least the first minor release—or even later. This approach usually allows library maintainers enough time to update their code.

**Comment by satanTime** [comment]
Hi all, happy New Year!

Please, let me finish holidays :) I'll review, update, merge and release next week.

**Comment by vinlos** [comment]
hi @satanTime

I understand it's hard to maintain an open source project alone. Do you accept pull requests for Angular major version update, so the community can contribute and speed up development?

**Comment by satanTime** [comment]
Hi @vinlos, any help is welcome.

**Comment by vladtheimpala** [comment]
big thanks for the updates, love ng-mocks

**Comment by Dji75** [comment]
Hi @satanTime hope you're well, nice to see you're back in the game ;)

For your new incoming developments, are you confident to support signal gracefully or not ? (do you expect something better than in https://github.com/help-me-mom/ng-mocks/issues/6918)
